### ISSUE TITLE
[Java][Client] 2.7.2 regression. MultiTopicsConsumer  org.apache.pulsar.client.api.PulsarClientException$NotSupportedException: not support shrink topic partitions

### ISSUE THREAD

**Issue by MichalKoziorowski-TomTom** [apache/pulsar#10717]
**Describe the bug**
After upgrading java client to 2.7.2 version we started to notice following logs:
```
2021-05-26 18:00:39.236 ERROR 11646 --- [r-client-io-1-1] o.a.p.c.impl.MultiTopicsConsumerImpl     : [persistent://public/batch/embedded_ROUTING1_bdcccf7eeaec7f5e0e7d243451433ac2ae8506d4] not support shrink topic partitions. old: 1, new: 0
2021-05-26 18:00:39.237  WARN 11646 --- [r-client-io-1-1] o.a.p.c.impl.MultiTopicsConsumerImpl     : [MultiTopicsConsumer-f03d6] Failed to subscribe increased topics partitions: org.apache.pulsar.client.api.PulsarClientException$NotSupportedException: not support shrink topic partitions
```

This happens couple of minutes of not pushing new messages to listened topic.
Another consequence of that are exceptions when trying to ack message that was in progress coming from that consumer:
```
org.apache.pulsar.client.api.PulsarClientException$InvalidMessageException: null
	at org.apache.pulsar.client.impl.ConsumerBase.acknowledgeAsync(ConsumerBase.java:373)
```

**Expected behavior**
There should be no errors.

**Additional context**
After upgrading server and java client to 2.7.2 version from 2.7.1 version, but it's problem in java client.

**Comment by MichalKoziorowski-TomTom** [comment]
I guess that issue can be somehow related to that change: https://github.com/apache/pulsar/pull/9660

**Comment by gaoran10** [comment]
It seems that the PR https://github.com/apache/pulsar/pull/10708 fixed the error log problem. I make some tests by the latest master and error logs `not support shrink topic partitions. old: 1, new: 0` are not present.

But I didn't reproduce the ack problem. Could you provide the steps that how to reproduce the ack problem?

**Comment by MichalKoziorowski-TomTom** [comment]
Hi @gaoran10 I haven't tested current master and it's quite hard to reproduce the issue.

**Comment by merlimat** [comment]
This should have been fixed by #10708

**Comment by eolivelli** [comment]
@merlimat this is a regression on branch-2.7.
#10708 is only for master branch (2.8.0...)

we should cherry pick to branch-2.7, correct ?

**Comment by MichalKoziorowski-TomTom** [comment]
Hi @merlimat, @eolivelli 

Can #10708 be cherry picked to new 2.7 release?

**Comment by eolivelli** [comment]
it is already marked to be cherry picked.

**Comment by MichalKoziorowski-TomTom** [comment]
Ahh. Ok. Thanks :)

**Comment by congbobo184** [comment]
https://github.com/apache/pulsar/pull/10708 has fix it and cherry-pick to 2.7, release 2.7.3, so close it.
